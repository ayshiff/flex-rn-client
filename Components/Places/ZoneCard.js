import { Card, CheckBox } from "react-native-elements";
import React from "react";
import I18n from "../../i18n/i18n";

const ZoneCard = (props: {
  checked: any,
  onPress: () => void,
  checked1: any,
  onPress1: () => void,
  checked2: any,
  onPress2: () => void,
}) => {
  const {
    checked,
    onPress,
    checked1,
    onPress1,
    checked2,
    onPress2,
  } = props;
  return (
    <Card>
      <CheckBox
        center
        title={I18n.t("places.red")}
        checkedIcon="dot-circle-o"
        checkedColor="#2E89AD"
        uncheckedIcon="circle-o"
        checked={checked}
        onPress={onPress}
      />
      <CheckBox
        center
        title={I18n.t("places.blue")}
        checkedIcon="dot-circle-o"
        checkedColor="#2E89AD"
        uncheckedIcon="circle-o"
        checked={checked1}
        onPress={onPress1}
      />
      <CheckBox
        center
        title={I18n.t("places.green")}
        checkedIcon="dot-circle-o"
        checkedColor="#2E89AD"
        uncheckedIcon="circle-o"
        checked={checked2}
        onPress={onPress2}
      />
    </Card>
  );
};

export default ZoneCard;
