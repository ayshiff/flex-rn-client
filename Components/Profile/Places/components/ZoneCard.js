import { Card, CheckBox } from "react-native-elements";
import React from "react";
import I18n from "../../../../i18n/i18n";

const ZoneCard = (props: {
  checked: any,
  onPress: () => void,
  checked1: any,
  onPress1: () => void,
  checked2: any,
  onPress2: () => void,
  checked3: any,
  onPress3: () => void
}) => {
  const {
    checked,
    onPress,
    checked1,
    onPress1,
    checked2,
    onPress2,
    checked3,
    onPress3
  } = props;
  return (
    <Card>
      <CheckBox
        center
        title={I18n.t("places.rer_zone")}
        checkedIcon="dot-circle-o"
        checkedColor="#2E89AD"
        uncheckedIcon="circle-o"
        checked={checked}
        onPress={onPress}
      />
      <CheckBox
        center
        title={I18n.t("places.forest_zone")}
        checkedIcon="dot-circle-o"
        checkedColor="#2E89AD"
        uncheckedIcon="circle-o"
        checked={checked1}
        onPress={onPress1}
      />
      <CheckBox
        center
        title={I18n.t("places.south_zone")}
        checkedIcon="dot-circle-o"
        checkedColor="#2E89AD"
        uncheckedIcon="circle-o"
        checked={checked2}
        onPress={onPress2}
      />
      <CheckBox
        center
        title={I18n.t("places.middle_zone")}
        checkedIcon="dot-circle-o"
        checkedColor="#2E89AD"
        uncheckedIcon="circle-o"
        checked={checked3}
        onPress={onPress3}
      />
    </Card>
  );
};

export default ZoneCard;
