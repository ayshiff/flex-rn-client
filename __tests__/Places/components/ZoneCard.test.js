import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import * as enzyme from "enzyme";
import ReactSixteenAdapter from "enzyme-adapter-react-16";
import ZoneCard from "../../../Components/Places/ZoneCard";

enzyme.configure({ adapter: new ReactSixteenAdapter() });

const onPress = jest.fn();
const onPress1 = jest.fn();
const onPress2 = jest.fn();

const checked = true;
const checked1 = false;
const checked2 = true;

it("renders correctly", () => {
  const wrapper = renderer
    .create(
      <ZoneCard
        onPress={onPress}
        onPress1={onPress1}
        onPress2={onPress2}
        checked={checked}
        checked1={checked1}
        checked2={checked2}
      />
    )
    .toJSON();
  expect(wrapper).toMatchSnapshot();
});
