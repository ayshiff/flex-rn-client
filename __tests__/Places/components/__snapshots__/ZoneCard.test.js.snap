// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderColor": "#ededed",
          "borderRadius": 3,
          "borderWidth": 1,
          "margin": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            false,
            Object {
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#2E89AD",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          [missing "fr.places.red" translation]
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderColor": "#ededed",
          "borderRadius": 3,
          "borderWidth": 1,
          "margin": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            false,
            Object {
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#bfbfbf",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          [missing "fr.places.blue" translation]
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderColor": "#ededed",
          "borderRadius": 3,
          "borderWidth": 1,
          "margin": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            false,
            Object {
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#2E89AD",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          [missing "fr.places.green" translation]
        </Text>
      </View>
    </View>
  </View>
</View>
`;
